client/src/components/Resources.js

```diff
@@ -78,195 +68,172 @@ export default class Resources extends Component<Props, State> {
+        setOpenKeys(latestOpenKey ? [latestOpenKey] : []);
+      }
+      const categorySelected = latestOpenKey;
+      props.history.push({
+        pathname: '/resources',
+        search: `?category=${categorySelected}`,
       });
-    }
-    const categorySelected = latestOpenKey;
-    this.props.history.push({
-      pathname: '/resources',
-      search: `?category=${categorySelected}`,
-    });
-  };
-
-  async updateResources() {
-    const [
-      categorySelected,
-      subcategorySelected,
-    ] = this.getCategorySelectedFromSearch();
-    const resources =
-      categorySelected === 'All Resources'
-        ? await getResources()
-        : await getResourcesByCategory(categorySelected);
-    this.setState({
-      categorySelected,
-      filteredResources: resources == null ? [] : resources.result,
-      openKeys: [categorySelected],
-      resources: resources == null ? [] : resources.result,
-      subcategorySelected,
-    });
-    this.filterResources({
-      cost: '$ - $$$$',
-      language: 'All',
-      location: 'All',
-      subcategory: subcategorySelected,
-      name: '',
-    });
-  }
-
-  render() {
-    const {
-      categories,
-      categorySelected,
-      costs,
-      costSelected,
-      filteredResources,
-      languages,
-      languageSelected,
-      locations,
-      locationSelected,
-      subcategorySelected,
-      openKeys,
-    } = this.state;
-
-    return (
-      <Layout className="resources">
-        <ResourcesBanner
-          categorySelected={categorySelected}
-          subcategorySelected={subcategorySelected}
-        />
-        <ResourcesFilter
-          costs={costs}
-          costSelected={costSelected}
-          languages={languages}
-          languageSelected={languageSelected}
-          locations={locations}
-          locationSelected={locationSelected}
-          handleChangeFilter={this.handleFilterChange}
-        />
-        <Layout style={{ background: 'white' }}>
-          <div>
-            <Sider className="filter-sider">
-              <Menu
-                mode="inline"
-                selectedKeys={
-                  subcategorySelected === ''
-                    ? categorySelected
-                    : subcategorySelected
-                }
-                openKeys={openKeys}
-                onOpenChange={this.onOpenChange}
+    },
+    [categories, category, openKeys, subcategory, props.history],
+  );
+
+  return (
+    <Layout className="resources">
+      <ResourcesBanner
+        categorySelected={category}
+        subcategorySelected={subcategory}
+      />
+      <ResourcesFilter
+        costs={costs}
+        costSelected={cost}
+        languages={languages}
+        languageSelected={language}
+        locations={locations}
+        locationSelected={location}
+        setCost={setCost}
+        setLanguage={setLanguage}
+        setLocation={setLocation}
+      />
+      <Layout style={{ background: 'white' }}>
+        <div>
```

Comment:
This can be a react fragment `<>`

Reply:
I believe if it's a fragment, the sidebar isn't sticky to the top anymore... not really sure why

Reply:
Oh weird - I remember this